### PR TITLE
TINY-7482: Added column header functionality

### PR DESCRIPTION
**Pull request by HAFRMO**
Related Ticket:

Description of Changes:
Added the column header ability.

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/plugins/table/main/ts/selection/SelectionTargets.ts [hunk 1]
```diff
@@ -163,12 +164,17 @@ export const getSelectionTargets = (editor: Editor, selections: Selections): Sel
     });
   };
 
-  const onSetupTableHeaders = (api: UiToggleApi): () => void => {
-    return onSetupWithToggle(api,
-      (targets) => isCaption(targets.element),
-      () => editor.queryCommandValue('mceTableRowType') === 'header'
-    );
-  };
+  const onSetupTableHeaders = (command: string, header: string) =>
```
Comment (JamesToohey) [inline modules/tinymce/src/plugins/table/main/ts/selection/SelectionTargets.ts:167]: Maybe?
```suggestion
  const onSetupTableHeaders = (command: string, headerType: 'header' | 'th') =>
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
+/* eslint-disable mocha/no-exclusive-tests */
```
Comment (JamesToohey) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:1]:
```suggestion
```

We have to be very careful adding this, it shouldn't be needed to run the tests with `.only` either
- [ ] Remove eslint comment

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+          '<tr>' +
+            makeCell(type, '1-0', scope) +
+            makeCell(type, '1-1', scope) +
+          '</tr>' +
+        '</tbody>' +
+      '</table>'
+    );
+  };
+
+  it('TINY-7481: Can toggle headers on with toolbar button', () => {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:60]:
```suggestion
  it('TINY-7481: Can toggle headers on with the toolbar button', () => {
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    const editor = hook.editor();
+    setEditorAndSelectionForOn(editor, 'td');
+
+    clickOnButton(editor, 'Column header');
+
+    TinyAssertions.assertContent(editor, getStructure('th'));
+  });
+
+  it('TINY-7481: Can toggle headers on with menuitem', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:69]:
```suggestion
  it('TINY-7481: Can toggle headers on with the menuitem', async () => {
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    const editor = hook.editor();
+    setEditorAndSelectionForOn(editor, 'td');
+
+    await pClickOnMenuItem(editor, 'Column header');
+
+    TinyAssertions.assertContent(editor, getStructure('th'));
+  });
+
+  it('TINY-7481: Can toggle headers off with toolbar button', () => {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:78]:
```suggestion
  it('TINY-7481: Can toggle headers off with the toolbar button', () => {
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    TinyAssertions.assertContent(editor, getStructure('th'));
+  });
+
+  it('TINY-7481: Can toggle headers on with menuitem', async () => {
+    const editor = hook.editor();
+    setEditorAndSelectionForOn(editor, 'td');
+
+    await pClickOnMenuItem(editor, 'Column header');
+
+    TinyAssertions.assertContent(editor, getStructure('th'));
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:75]:
```suggestion
    await pClickOnMenuItem(editor, 'Column header');
    TinyAssertions.assertContent(editor, getStructure('th'));
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    );
+  };
+
+  it('TINY-7481: Can toggle headers on with toolbar button', () => {
+    const editor = hook.editor();
+    setEditorAndSelectionForOn(editor, 'td');
+
+    clickOnButton(editor, 'Column header');
+
+    TinyAssertions.assertContent(editor, getStructure('th'));
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:66]:
```suggestion
    clickOnButton(editor, 'Column header');
    TinyAssertions.assertContent(editor, getStructure('th'));
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  });
+
+  it('TINY-7481: Can toggle headers off with toolbar button', () => {
+    const editor = hook.editor();
+    setEditorAndSelectionForOn(editor, 'th');
+
+    clickOnButton(editor, 'Column header');
+
+    TinyAssertions.assertContent(editor, getStructure('td'));
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:84]:
```suggestion
    clickOnButton(editor, 'Column header');
    TinyAssertions.assertContent(editor, getStructure('td'));
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  });
+
+  it('TINY-7481: Can toggle headers off with menuitem', async () => {
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:87]:
```suggestion
  it('TINY-7481: Can toggle headers off with the menuitem', async () => {
```

File: modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+    const editor = hook.editor();
+    setEditorAndSelectionForOn(editor, 'th');
+
+    await pClickOnMenuItem(editor, 'Column header');
+
+    TinyAssertions.assertContent(editor, getStructure('td'));
```
Comment (lnewson) [inline modules/tinymce/src/plugins/table/test/ts/browser/ui/TableColumnHeaderUiTest.ts:93]:
```suggestion
    await pClickOnMenuItem(editor, 'Column header');
    TinyAssertions.assertContent(editor, getStructure('td'));
```

File: modules/tinymce/CHANGELOG.md [hunk 11]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - A new `lang` format is now available that can be used with `editor.formatter`, or applied with the `Lang` editor command #TINY-6149
 - Added new plugin commands: `mceEmoticons` (emoticons), `mceWordCount` (wordcount), and `mceTemplate` (template) #TINY-7619
 - Added a new `tablerowheader` toolbar button and menu item to toggle row cells to and from header #TINY-7478
+- Added a new `tablecolheader` toolbar button and menu item to toggle row cells to and from header #TINY-7482
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:26]:
```suggestion
- Added a new `tablecolheader` toolbar button and menu item to toggle column cells header state #TINY-7482
```
Maybe? This at the very least needs to say column, not row.

**Review by lnewson — APPROVED** [review]
Looks good, just mostly grammar/general cleanup.

**Review by ashfordneil — APPROVED** [review]

**Review by ltrouton — APPROVED** [review]

**Review by JamesToohey — APPROVED** [review]

File: modules/tinymce/CHANGELOG.md (hunk 11 above)
Comment (lnewson) [reply modules/tinymce/CHANGELOG.md]: This still reads really strangely (so does the row one for that matter). I don't think we need the `to and from` part when we're already saying `toggles`

Comment (lnewson) [reply modules/tinymce/CHANGELOG.md]:
```
- Added a new `tablecolheader` toolbar button and menu item to toggle the header state of column cells #TINY-7482
```
This is probably better than my original suggestion.

File: modules/tinymce/CHANGELOG.md [hunk 12]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - A new `lang` format is now available that can be used with `editor.formatter`, or applied with the `Lang` editor command #TINY-6149
 - Added new plugin commands: `mceEmoticons` (emoticons), `mceWordCount` (wordcount), and `mceTemplate` (template) #TINY-7619
 - Added a new `tablerowheader` toolbar button and menu item to toggle row cells to and from header #TINY-7478
+- Added a new `tablecolheader` toolbar button and menu item to toggle column cells to and from header state #TINY-7482
```
Comment (lnewson) [inline modules/tinymce/CHANGELOG.md:26]:
```suggestion
- Added a new `tablerowheader` toolbar button and menu item to toggle the header state of row cells #TINY-7478
- Added a new `tablecolheader` toolbar button and menu item to toggle the header state of column cells #TINY-7482
```
This is probably better than my original suggestion.

**Pull request merged**